test/tests/functional/pbs_snapshot_unittest.py

```diff
@@ -97,9 +98,17 @@ def setup_sc(self, sched_id, partition, port):
         a = {'partition': partition,
              'sched_host': self.server.hostname,
              'sched_port': port}
+        if sched_priv is not None:
+            a.update({'sched_priv': sched_priv})
```

Comment:
this is inefficient, please replace this with simply a['sched_priv'] = sched_priv. Same for sched_log.